src/OSInfo.h

```diff
@@ -18,6 +18,24 @@ class OSInfo {
 		static QString getMacHash(const QList<QHostAddress> & = QList<QHostAddress>());
 		static QString getOS();
 		static QString getOSVersion();
+#if defined(Q_OS_WIN)
+		static QString getOSVersionWindows();
+#elif defined(Q_OS_MAC)
+		static QString getOSVersionMac();
+#else
+		static QString getOSVersionOther();
+		static QString getOSVersionOtherUname();
+#endif
+#ifdef Q_OS_LINUX
+		static QString getOSVersionLinux();
+		/// Determine OS as per systemd os-release file
+		/// Use PRETTY_NAME if available, otherwise NAME
+		/// systemd defines that if no NAME is defined the default is "Linux", but this method will
+		/// return an empty string (because we will fall back to other methods of determining the version).
+		/// https://www.freedesktop.org/software/systemd/man/os-release.html
+		static QString getOSVersionLinuxOsrelease();
+		static QString getOSVersionLinuxLsbrelease();
+#endif
```

Comment:
Doxygen-comments for the other functions would be nice as well. Maybe this could contain a short info about where this function looks for the OS info...